trading_cost.py

```diff
@@ -0,0 +1,64 @@
+"""
+A file to build cost objects.
+"""
+
+from collections import defaultdict
+
+class CurrencyCost:
+    """
+    Cost function of currency trading.
+    """
+    def __init__(self, base_ins, instruments):
+        self.base_ins = base_ins
+        self.instruments = instruments
+        self.quantity = defaultdict(int)
+        self.value = defaultdict(int)
+        self.value[base_ins] = 1
+        self.quantity[base_ins] = 1
+
+    def __call__(self, portfolio, price, liquidation):
+        if liquidation:
+            for instrument in self.instruments:
+                portfolio[-1][instrument] = (
+                    0 if instrument is not self.base_ins else 1)
+        return self.calculate(portfolio[-1], price)
+
+    def calculate(self, today, price):
+        """
+        Calculate pnl and produce cost.
+
+        We assume that our investing money is maintained in 1.
+        That make it more general to evaluate the perfoormance for the alpha.
+        For detailed, please read our wiki pages.
+        """
+        pnl = {
+            instrument : 0 for instrument in self.instruments}
+        cost = {
+            instrument : 0 for instrument in self.instruments}
```

Comment:
Can we make both them in a single line?
I think their width should be <= 80.
Besides, there are redundant spaces between `instrument` and `:`.

Reply:
After thinking, maybe we should default the value of keys which we don't get to 0 ? 
If we don't do that, we can't settle all account.
ex.
10/01 USD: 0.5, JPY: 0.5
10/02 USD: 0.5, EUR: 0.5
That may means we sell JPY and buy EUR. 
If we didn't default portfolio of JPY to 0 on 10/02, the JPY account will remain 0.5 (due to the operation in line 39).
I think set them to 0 may be an easier way to fix this problem, and this method can be build in portfolio.py maybe ?.

Reply:
Well, what about modifying line 39 directly?
We can check all instruments there, and move the tradable test into Portfolio.
Of course we can keep a set which stores "non-zero instruments" to optimize the performance, but it will make this CL hard to read. So let's do it in the future CLs.